packages/manager/modules/anthos/src/dashboard/ips/remove-private-ip/remove-private-ip.controller.js

```diff
@@ -0,0 +1,50 @@
+export default class RemovePrivateIpController {
+  /* @ngInject */
+  constructor($translate, Alerter, AnthosTenantsService) {
+    this.$translate = $translate;
+    this.Alerter = Alerter;
+    this.AnthosTenantsService = AnthosTenantsService;
+  }
+
+  removePrivateIp(rangeId) {
+    const privateIpIndex = this.privateIPs.findIndex(
+      ({ id }) => id === rangeId,
+    );
+    if (privateIpIndex >= 0) {
+      this.privateIPs.splice(privateIpIndex, 1);
+    }
+  }
+
+  onRemovePrivateIpClick() {
+    this.isRemoving = true;
+
+    return this.AnthosTenantsService.removeTenantPrivateIP(
+      this.serviceName,
+      this.privateIp.id,
+    )
+      .then(() => {
+        this.removePrivateIp(this.privateIp.id);
```

Comment:
Since `removePrivateIp` being called both in then and catch, we can move this to `finally` block

Reply:
It's an issue on catch block.
Thanks for observation